common/sampling.h

```diff
@@ -14,6 +14,7 @@ typedef struct llama_sampling_params {
     int32_t n_probs           = 0;     // if greater than 0, output the probabilities of top n_probs tokens.
     int32_t top_k             = 40;    // <= 0 to use vocab size
     float   top_p             = 0.95f; // 1.0 = disabled
+    float   min_p             = 0.00f; // 0.0 = disabled
```

Comment:
When min_p is non-positive, it is expected that sampling will be greedy, i.e. only min_keep logits will be kept (1 in our case). When it is greater or equal to 1, it is expected that it is disabled.

Reply:
This parameter represents a minimum, whereas parameters like top-p represent a maximum. Should it really be the case that for a given *p*, the sampler rejects tokens with a probability less than 1-*p*?

Reply:
Sorry, I was wrong. I thought about it like "minimum probability threshold", but like you said in inverted `1-p` case.